Source/Applications/SystemCenter/Controllers/SystemCenter/SystemCenterController.cs

```diff
@@ -840,80 +839,112 @@ private IEnumerable<ParsedChannel> ParsePQDiff(byte[] bytes, string meterKey)
                     .ToList();
 
-                // Create the list of series instances so we can
-                // build it as we process each channel instance
-                List<SeriesInstance> seriesInstances = new List<SeriesInstance>();
-
-                List<openXDA.Model.Channel> parsedChannels = new List<openXDA.Model.Channel>();
+                List<ParsedChannel> parsedChannels = new List<ParsedChannel>();
+                int channelIndex = 1;
+                bool assumeTrend = true;
                 foreach (ChannelInstance channelInstance in channelInstances)
                 {
                     bool timeValueChannel =
                         channelInstance.Definition.QuantityTypeID == QuantityType.WaveForm ||
                         channelInstance.Definition.QuantityTypeID == QuantityType.ValueLog ||
                         channelInstance.Definition.QuantityTypeID == QuantityType.Phasor;
 
-                    if (!timeValueChannel || channelInstance.SeriesInstances.Count() == 0)
+                    if (!timeValueChannel)
                         continue;
 
-                    openXDA.Model.Channel channel = PQDIFReader.ParseSeries(channelInstance.SeriesInstances[0]);
-                    channel.Series = channelInstance.SeriesInstances.SelectMany((seriesInstance) =>
-                        {
-                            seriesInstances.Add(seriesInstance);
-                            openXDA.Model.Channel tempChannel = PQDIFReader.ParseSeries(seriesInstance);
-                            return tempChannel.Series;
-                        }).ToList();
+                    ParsedChannel channel = ParseChannel(channelInstance, assumeTrend);
+                    channel.Meter = meterKey;
+                    channel.ID = channelIndex;
+                    channelIndex++;
                     // Add the new channel to the meter's channel list
                     parsedChannels.Add(channel);
                 }
+                return parsedChannels;
+            }
+        }
 
-                // Build a list of series definitions that were not instanced by this PQDIF file
-                List<SeriesDefinition> seriesDefinitions = dataSources
-                    .SelectMany(dataSource => dataSource.ChannelDefinitions)
-                    .SelectMany(channelDefinition => channelDefinition.SeriesDefinitions)
-                    .Distinct()
-                    .Except(seriesInstances.Select(seriesInstance => seriesInstance.Definition))
-                    .ToList();
+        private static ParsedChannel ParseChannel(ChannelInstance channelInstance, bool lastTrend = true)
+        {
+            // Populate measurement type properties
+            QuantityMeasured quantityMeasured = channelInstance.Definition.QuantityMeasured;
+
+            // Popuplate phase properties
+            Phase phase = channelInstance.Definition.Phase;
```

Comment:
Generally Ok, but where this will fail is if it encounters a new Phase that is not in the Phase Table yet. XDA had the ability to auto add `Phase`, `MeasurementType` and `MeaurementCharacteristic` to the Tables if they don't exist.
As long as the NMW can handle that in TSX that's fine, but you will need to confirm that.